lib/medium-admin.php

```diff
@@ -1030,19 +1043,25 @@ private static function _handle_response($response) {
     $content_type = wp_remote_retrieve_header($response, "content-type");
     $body = wp_remote_retrieve_body($response);
 
+    error_log("Received response ($code - $content_type): $body");
     if (is_wp_error($response)) {
-      throw new Exception($response->get_error_message(), 500);
+      $err_msg = $response->get_error_message();
+      $err_code = $response->get_error_code();
+      error_log("WP ERROR: $err_msg ($err_code)");
+      if ($err_code == "http_request_failed" && strpos($err_msg, "timed out") !== false) {
+        throw new Exception($err_msg, 1001); // our custom code for timeouts
+      }
+      throw new Exception($err_msg, 500);
     }
 
     if (false === strpos($content_type, "json")) {
       throw new Exception(__("Unexpected response format.", "medium"), $code);
     }
 
-    error_log("Received payload: $body");
     $payload = json_decode($body);
     if (isset($payload->errors)) {
       $error = $payload->errors[0];
-      error_log("Received API error: $error->message ($error->code)");
+      error_log("API ERROR: $error->message ($error->code)");
```

Comment:
We're logging a bunch of stuff to error logs, so this helps to pick it out in a `tail -f`. TBH we'll want a much better/more sustainable logging solution before cutting a new version anyway, this is just for the pending migration